extension/fts/src/function/query_fts_bind_data.cpp

```diff
@@ -0,0 +1,100 @@
+#include "function/query_fts_bind_data.h"
+
+#include "binder/expression/expression_util.h"
+#include "catalog/fts_index_catalog_entry.h"
+#include "common/string_utils.h"
+#include "function/fts_utils.h"
+#include "function/stem.h"
+#include "libstemmer.h"
+#include "re2.h"
+#include "storage/storage_manager.h"
+#include "storage/store/node_table.h"
+
+namespace kuzu {
+namespace fts_extension {
+
+using namespace kuzu::common;
+using namespace kuzu::binder;
+using namespace kuzu::storage;
+
+static std::string evaluateQuery(const Expression& query) {
+    if (!ExpressionUtil::canEvaluateAsLiteral(query)) {
+        std::string errMsg;
+        switch (query.expressionType) {
+        case ExpressionType::PARAMETER: {
+            errMsg = "The query is a parameter expression. Please assign it a value.";
+        } break;
+        default: {
+            errMsg = "The query must be a parameter/literal expression.";
+        } break;
+        }
+        throw RuntimeException{errMsg};
+    }
+    auto value = binder::ExpressionUtil::evaluateAsLiteralValue(query);
+    if (value.getDataType() != common::LogicalType::STRING()) {
+        throw RuntimeException{"The query must be a string literal."};
+    }
+    return value.getValue<std::string>();
+}
+
+static void normalizeQuery(std::string& query) {
+    std::string regexPattern = "[0-9!@#$%^&*()_+={}\\[\\]:;<>,.?~\\/\\|'\"`-]+";
+    std::string replacePattern = " ";
+    RE2::GlobalReplace(&query, regexPattern, replacePattern);
+    StringUtils::toLower(query);
+}
+
+struct StopWordsChecker {
+    ValueVector termsVector;
+    common::offset_t offset = common::INVALID_OFFSET;
+    storage::NodeTable* stopWordsTable;
+    transaction::Transaction* tx;
+
+    explicit StopWordsChecker(main::ClientContext& context);
+    bool isStopWord(const std::string& term);
+};
+
+StopWordsChecker::StopWordsChecker(main::ClientContext& context)
+    : termsVector{LogicalType::STRING(), context.getMemoryManager()}, tx{context.getTransaction()} {
+    termsVector.state = common::DataChunkState::getSingleValueDataChunkState();
+    auto tableID = context.getCatalog()->getTableID(tx, FTSUtils::getStopWordsTableName());
+    stopWordsTable = context.getStorageManager()->getTable(tableID)->ptrCast<storage::NodeTable>();
+}
+
+bool StopWordsChecker::isStopWord(const std::string& term) {
```

Comment:
Since this is a cpp file. I would just inline implementation with declaration. But this is up to u